src/util/StringUtils.h

```diff
@@ -132,7 +132,17 @@ std::pair<size_t, std::string> getUTF8Prefix(std::string_view sv,
           "Illegal UTF sequence in ad_utility::getUTF8Prefix");
     }
   }
-  return {numCodepoints, std::string(sv.data(), i)};
+  return {numCodepoints, sv.substr(0, i)};
+}
+
+inline string_view getUTF8Substring(const std::string_view str, size_t start,
+                                    size_t size) {
+  auto [numCodepoints, upToEnd] = getUTF8Prefix(str, start + size);
+  auto [numCodepoints2, prefix] = getUTF8Prefix(upToEnd, start);
+  return upToEnd.substr(prefix.size());
+}
+inline string_view getUTF8Substring(const std::string_view str, size_t start) {
+  return getUTF8Substring(str, start, str.size());
 }
```

Comment:
```suggestion
// Overload for the above function that creates the substring from the `start`-th codepoint to the end of the string.
inline string_view getUTF8Substring(const std::string_view str, size_t start) {
  // `str.size()` is >= the number of codepoints because each codepoint has at least one byte in UTF-8
  return getUTF8Substring(str, start, str.size());
}
```